File: Plugins/VanillaBasics/src/main/java/org/tobi29/scapes/vanilla/basics/VanillaBasicsOres.java
/*
 * Copyright 2012-2016 Tobi29
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package org.tobi29.scapes.vanilla.basics;

import org.tobi29.scapes.vanilla.basics.material.StoneType;
import org.tobi29.scapes.vanilla.basics.material.VanillaMaterial;

class VanillaBasicsOres {
    static void registerOres(VanillaBasics plugin) {
        VanillaMaterial materials = plugin.getMaterials();
        plugin.c.ore(o -> {
            o.type = materials.oreCoal;
            o.rarity = 9;
            o.size = 16.0;
            o.chance = 3;
            o.rockChance = 20;
            o.rockDistance = 16;
            o.stoneTypes.add(StoneType.DIRT_STONE);
        });
        plugin.c.ore(o -> {
            o.type = materials.oreCoal;
            o.rarity = 3;
            o.size = 12.0;
            o.chance = 3;
            o.rockChance = 20;
            o.rockDistance = 32;
            o.stoneTypes.add(StoneType.CHALK);
            o.stoneTypes.add(StoneType.CHERT);
            o.stoneTypes.add(StoneType.CLAYSTONE);
            o.stoneTypes.add(StoneType.CONGLOMERATE);
            o.stoneTypes.add(StoneType.MARBLE);
            o.stoneTypes.add(StoneType.ANDESITE);
            o.stoneTypes.add(StoneType.BASALT);
            o.stoneTypes.add(StoneType.DACITE);
            o.stoneTypes.add(StoneType.RHYOLITE);
            o.stoneTypes.add(StoneType.DIORITE);
            o.stoneTypes.add(StoneType.GABBRO);
            o.stoneTypes.add(StoneType.GRANITE);
        });
        plugin.c.ore(o -> {
            o.type = materials.oreCassiterite;
            o.rarity = 2;
            o.size = 24.0;
            o.chance = 64;
            o.rockChance = 12;
            o.rockDistance = 32;
            o.stoneTypes.add(StoneType.ANDESITE);
            o.stoneTypes.add(StoneType.BASALT);
            o.stoneTypes.add(StoneType.DACITE);
            o.stoneTypes.add(StoneType.RHYOLITE);
            o.stoneTypes.add(StoneType.GRANITE);
        });
        plugin.c.ore(o -> {
            o.type = materials.oreSphalerite;
            o.rarity = 6;
            o.size = 6.0;
            o.chance = 3;
            o.rockChance = 4;
            o.rockDistance = 16;
            o.stoneTypes.add(StoneType.MARBLE);
        });
        plugin.c.ore(o -> {
            o.type = materials.oreBismuthinite;
            o.rarity = 2;
            o.size = 3.0;
            o.chance = 8;
            o.rockChance = 9;
            o.rockDistance = 128;
            o.stoneTypes.add(StoneType.CHALK);
            o.stoneTypes.add(StoneType.CHERT);
            o.stoneTypes.add(StoneType.CLAYSTONE);
            o.stoneTypes.add(StoneType.CONGLOMERATE);
            o.stoneTypes.add(StoneType.MARBLE);
            o.stoneTypes.add(StoneType.DIORITE);
            o.stoneTypes.add(StoneType.GABBRO);
            o.stoneTypes.add(StoneType.GRANITE);
        });
        plugin.c.ore(o -> {
            o.type = materials.oreChalcocite;
            o.rarity = 12;
            o.size = 8.0;
            o.chance = 2;
            o.rockChance = 1;
            o.rockDistance = 24;
            o.stoneTypes.add(StoneType.CHALK);
            o.stoneTypes.add(StoneType.CHERT);
            o.stoneTypes.add(StoneType.CLAYSTONE);
            o.stoneTypes.add(StoneType.CONGLOMERATE);
        });
        plugin.c.ore(o -> {
            o.type = materials.oreMagnetite;
            o.rarity = 4;
            o.size = 64.0;
            o.chance = 12;
            o.rockChance = 10;
            o.rockDistance = 96;
            o.stoneTypes.add(StoneType.CHALK);
            o.stoneTypes.add(StoneType.CHERT);
            o.stoneTypes.add(StoneType.CLAYSTONE);
            o.stoneTypes.add(StoneType.CONGLOMERATE);
        });
        plugin.c.ore(o -> {
            o.type = materials.orePyrite;
            o.rarity = 3;
            o.size = 4.0;
            o.chance = 11;
            o.rockChance = 13;
            o.rockDistance = 48;
            o.stoneTypes.add(StoneType.CHALK);
            o.stoneTypes.add(StoneType.CHERT);
            o.stoneTypes.add(StoneType.CLAYSTONE);
            o.stoneTypes.add(StoneType.CONGLOMERATE);
            o.stoneTypes.add(StoneType.MARBLE);
        });
        plugin.c.ore(o -> {
            o.type = materials.oreSilver;
            o.rarity = 2;
            o.size = 3.0;
            o.chance = 4;
            o.rockChance = 8;
            o.rockDistance = 64;
            o.stoneTypes.add(StoneType.GRANITE);
        });
        plugin.c.ore(o -> {
            o.type = materials.oreGold;
            o.rarity = 1;
            o.size = 2.0;
            o.chance = 4;
            o.rockChance = 96;
            o.rockDistance = 256;
            o.stoneTypes.add(StoneType.ANDESITE);
            o.stoneTypes.add(StoneType.BASALT);
            o.stoneTypes.add(StoneType.DACITE);
            o.stoneTypes.add(StoneType.RHYOLITE);
            o.stoneTypes.add(StoneType.DIORITE);
            o.stoneTypes.add(StoneType.GABBRO);
            o.stoneTypes.add(StoneType.GRANITE);
        });
    }
}

File: Plugins/VanillaBasics/src/main/kotlin/scapes/plugin/tobi29/vanilla/basics/VanillaBasicsOres.kt
/*
 * Copyright 2012-2016 Tobi29
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package scapes.plugin.tobi29.vanilla.basics

import scapes.plugin.tobi29.vanilla.basics.material.StoneType

internal fun VanillaBasics.registerOres() {
    ore {
        type = materials.oreCoal
        rarity = 9
        size = 16.0
        chance = 3
        rockChance = 20
        rockDistance = 16
        stoneTypes.add(StoneType.DIRT_STONE)
    }
    ore {
        type = materials.oreCoal
        rarity = 3
        size = 12.0
        chance = 3
        rockChance = 20
        rockDistance = 32
        stoneTypes.add(StoneType.CHALK)
        stoneTypes.add(StoneType.CHERT)
        stoneTypes.add(StoneType.CLAYSTONE)
        stoneTypes.add(StoneType.CONGLOMERATE)
        stoneTypes.add(StoneType.MARBLE)
        stoneTypes.add(StoneType.ANDESITE)
        stoneTypes.add(StoneType.BASALT)
        stoneTypes.add(StoneType.DACITE)
        stoneTypes.add(StoneType.RHYOLITE)
        stoneTypes.add(StoneType.DIORITE)
        stoneTypes.add(StoneType.GABBRO)
        stoneTypes.add(StoneType.GRANITE)
    }
    ore {
        type = materials.oreCassiterite
        rarity = 2
        size = 24.0
        chance = 64
        rockChance = 12
        rockDistance = 32
        stoneTypes.add(StoneType.ANDESITE)
        stoneTypes.add(StoneType.BASALT)
        stoneTypes.add(StoneType.DACITE)
        stoneTypes.add(StoneType.RHYOLITE)
        stoneTypes.add(StoneType.GRANITE)
    }
    ore {
        type = materials.oreSphalerite
        rarity = 6
        size = 6.0
        chance = 3
        rockChance = 4
        rockDistance = 16
        stoneTypes.add(StoneType.MARBLE)
    }
    ore {
        type = materials.oreBismuthinite
        rarity = 2
        size = 3.0
        chance = 8
        rockChance = 9
        rockDistance = 128
        stoneTypes.add(StoneType.CHALK)
        stoneTypes.add(StoneType.CHERT)
        stoneTypes.add(StoneType.CLAYSTONE)
        stoneTypes.add(StoneType.CONGLOMERATE)
        stoneTypes.add(StoneType.MARBLE)
        stoneTypes.add(StoneType.DIORITE)
        stoneTypes.add(StoneType.GABBRO)
        stoneTypes.add(StoneType.GRANITE)
    }
    ore {
        type = materials.oreChalcocite
        rarity = 12
        size = 8.0
        chance = 2
        rockChance = 1
        rockDistance = 24
        stoneTypes.add(StoneType.CHALK)
        stoneTypes.add(StoneType.CHERT)
        stoneTypes.add(StoneType.CLAYSTONE)
        stoneTypes.add(StoneType.CONGLOMERATE)
    }
    ore {
        type = materials.oreMagnetite
        rarity = 4
        size = 64.0
        chance = 12
        rockChance = 10
        rockDistance = 96
        stoneTypes.add(StoneType.CHALK)
        stoneTypes.add(StoneType.CHERT)
        stoneTypes.add(StoneType.CLAYSTONE)
        stoneTypes.add(StoneType.CONGLOMERATE)
    }
    ore {
        type = materials.orePyrite
        rarity = 3
        size = 4.0
        chance = 11
        rockChance = 13
        rockDistance = 48
        stoneTypes.add(StoneType.CHALK)
        stoneTypes.add(StoneType.CHERT)
        stoneTypes.add(StoneType.CLAYSTONE)
        stoneTypes.add(StoneType.CONGLOMERATE)
        stoneTypes.add(StoneType.MARBLE)
    }
    ore {
        type = materials.oreSilver
        rarity = 2
        size = 3.0
        chance = 4
        rockChance = 8
        rockDistance = 64
        stoneTypes.add(StoneType.GRANITE)
    }
    ore {
        type = materials.oreGold
        rarity = 1
        size = 2.0
        chance = 4
        rockChance = 96
        rockDistance = 256
        stoneTypes.add(StoneType.ANDESITE)
        stoneTypes.add(StoneType.BASALT)
        stoneTypes.add(StoneType.DACITE)
        stoneTypes.add(StoneType.RHYOLITE)
        stoneTypes.add(StoneType.DIORITE)
        stoneTypes.add(StoneType.GABBRO)
        stoneTypes.add(StoneType.GRANITE)
    }
}
